### PR TITLE
[cucumber] Adds BeforeAll and AfterAll to types

### PR DESCRIPTION
**Pull request by ath88**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/hooks.md#beforeall--afterall
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Not much else to say.

### CODE REVIEW

**Comment by kevinkuszyk** [comment]
AbraaoAlves @jan-molak @isoung @BendingBender do you have an ETA for merging this?

I have some additions, but they would be better based off this rather than the `master`.

File: types/cucumber/index.d.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-// Type definitions for cucumber-js 2.1
+// Type definitions for cucumber-js 3.1
 // Project: https://github.com/cucumber/cucumber-js
 // Definitions by: Abraão Alves <https://github.com/abraaoalves>
 //                 Jan Molak <https://github.com/jan-molak>
 //                 Isaiah Soung <https://github.com/isoung>
 //                 BendingBender <https://github.com/BendingBender>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.4
```
Comment (minestarks) [inline types/cucumber/index.d.ts]: It doesn't look like you're adding anything that requires a minimum TS version of 2.4. Could you revert this?

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/cucumber/index.d.ts (hunk 1 above)
Comment (aseeeem) [reply types/cucumber/index.d.ts]: :eyes:

Comment (ath88) [reply types/cucumber/index.d.ts]: Will do tomorrow, and you're right. My bad.

Comment (aseeeem) [reply types/cucumber/index.d.ts]: Opened a PR on your fork @ath88!
Ran the tests and such as well. 

If you have time today, that would be _sweet_

https://github.com/ath88/DefinitelyTyped/pull/1

Comment (ath88) [reply types/cucumber/index.d.ts]: Thanks! Done.

Comment (aseeeem) [reply types/cucumber/index.d.ts]: No problem :smile_cat: Thanks for checking it in!

**Comment by aseeeem** [comment]
@minestarks When you get a chance, please review the changes!

Thanks!

**Review by minestarks — APPROVED** [review]

**Comment by aseeeem** [comment]
:raised_hands: thanks @minestarks !

**Pull request merged**